src/core/lib/backoff/backoff.cc

```diff
@@ -34,9 +34,7 @@ void grpc_backoff_init(grpc_backoff *backoff,
 
 grpc_millis grpc_backoff_begin(grpc_exec_ctx *exec_ctx, grpc_backoff *backoff) {
   backoff->current_timeout_millis = backoff->initial_connect_timeout;
-  const grpc_millis first_timeout =
-      GPR_MAX(backoff->current_timeout_millis, backoff->min_timeout_millis);
-  return grpc_exec_ctx_now(exec_ctx) + first_timeout;
+  return grpc_exec_ctx_now(exec_ctx) + backoff->current_timeout_millis;
```

Comment:
The variable `current_timeout_millis` should be actually `current_backoff_millis`. So is `max_timeout_millis`. Leaving them as 'timeout' messes up with readability and error prone.